### PR TITLE
mgmt/mcumgr/lib: Cleanups in img management group

### PR DESCRIPTION
**Pull request by de-nordic**
Two commits:
 1) Removal of unfinished event logging via SMP protocol - except from some callbacks and vague SMP fram descriptions nothing has been implemented;
 2) Merging private headers of img management and removing zephyr_ and _impl_ from function names to reduce name length.

- [x] Testing

### CODE REVIEW

File: subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.h [hunk 1]
```diff
@@ -4,59 +4,138 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+/**
+ * @file
+ * @brief Declares implementation-specific functions required by image management.  The default
+ * stubs can be overridden with functions that are compatible with the host OS.
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.h:10]: These aren't weak functions and can't be overridden

**Review by nordicjm — COMMENTED** [review]
Typo in commit message `The commit removes zephry_`

File: subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.h (hunk 1 above)
Comment (de-nordic) [reply subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.h:10]: They never were. I should fix the doc.

File: subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.h [hunk 2]
```diff
@@ -4,59 +4,138 @@
...
+/**
+ * @brief Writes the specified chunk of image data to slot 1.
  *
- * Request to boot to version:
- * {
- *	"test":<version>
- * }
+ * @param offset	The offset within slot 1 to write to.
+ * @param data		The image data to write.
+ * @param num_bytes	The number of bytes to read.
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.h:69]: *write

File: subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.h [hunk 3]
```diff
@@ -4,59 +4,138 @@
...
+ */
+int img_mgmt_write_image_data(unsigned int offset, const void *data, unsigned int num_bytes,
+			      bool last);
+
+/**
+ * @brief Indicates the type of swap operation that will occur on the next
+ * reboot, if any, between provided slot and it's pair.
+ * Querying any slots of the same pair will give the same result.
+ *
+ * @param image                 An slot number;
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.h:84]: *A slot

File: subsys/mgmt/mcumgr/lib/cmd/img_mgmt/CMakeLists.txt [hunk 4]
```diff
@@ -9,8 +9,8 @@ target_include_directories(MCUMGR INTERFACE
 )
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/lib/cmd/img_mgmt/CMakeLists.txt:9]: No issue on this line, however typo in commit message *zephyr_

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: subsys/mgmt/mcumgr/lib/cmd/img_mgmt/CMakeLists.txt (hunk 4 above)
Comment (nordicjm) [reply subsys/mgmt/mcumgr/lib/cmd/img_mgmt/CMakeLists.txt:9]: Typo still present:

> The commit removes zephry_ and _impl_ from

File: subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.c [hunk 5]
```diff
@@ -618,6 +614,8 @@ img_mgmt_impl_upload_inspect(const struct img_mgmt_upload_req *req,
 		}
 
 #ifndef CONFIG_IMG_ERASE_PROGRESSIVELY
+		(void) empty;
+#else
 		rc = img_mgmt_flash_check_empty(action->area_id, &empty);
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.c:619]: This seems to be reversed

File: subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.c [hunk 6]
```diff
@@ -96,7 +95,7 @@ static int img_mgmt_flash_check_empty_inner(const struct flash_area *fa, bool *o
 	return 0;
 }
 
-#ifndef CONFIG_IMG_ERASE_PROGRESSIVELY
+#ifdef CONFIG_IMG_ERASE_PROGRESSIVELY
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.c:98]: This seems to be reversed

File: subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.h (hunk 3 above)
Comment (nordicjm) [reply subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.h:86]: Needs correcting

File: subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.h (hunk 2 above)
Comment (nordicjm) [reply subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.h:71]: Needs correcting

File: subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.h (hunk 1 above)
Comment (nordicjm) [reply subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_priv.h:10]: Comment about weak stubs can go

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Review by nvlsianpu — DISMISSED** [review]

File: subsys/mgmt/mcumgr/lib/cmd/img_mgmt/CMakeLists.txt [hunk 7]
```diff
@@ -9,9 +9,8 @@ target_include_directories(MCUMGR INTERFACE
...
 
 zephyr_library_sources(
-    src/zephyr_img_mgmt.c
```
[the remark this answers is not in the document]
Comment (de-nordic) [reply subsys/mgmt/mcumgr/lib/cmd/img_mgmt/CMakeLists.txt:12]: This is img_mgmt_priv.c now.

**Review by nordicjm — DISMISSED** [review]

**Review by nordicjm — APPROVED** [review]

**Comment by de-nordic** [comment]
@carlescufi @nvlsianpu Can you take a look?

**Review by carlescufi — APPROVED** [review]

**Pull request merged**